resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/DefaultResponseExceptionMapper.java

```diff
@@ -13,12 +18,15 @@ public Throwable toThrowable(Response response) {
         try {
             response.bufferEntity();
         } catch (Exception ignored) {}
-        return new WebApplicationException("Unknown error, status code " + response.getStatus(), response);
+        return WebApplicationExceptionWrapper.wrap(new WebApplicationException("Unknown error, status code " + response.getStatus(), response));
     }
 
     @Override
     public boolean handles(int status, MultivaluedMap headers) {
-        return status >= 400;
+       final Config config = ConfigProvider.getConfig();
+       final boolean originalBehavior = config.getOptionalValue(ResteasyContextParameters.RESTEASY_ORIGINAL_WEBAPPLICATIONEXCEPTION_BEHAVIOR, boolean.class).orElse(false);
+       final boolean serverSide = ResteasyDeployment.onServer();
+       return status >= (originalBehavior || !serverSide ? 400 : 300);
```

Comment:
This looks different from how we did it in the 3.x branches.

Reply:
I realized as I was releasing 3.11.3.Final that I had changed the test to include 3xx statuses, which makes sense, I think, since the same leakage problem applies. There was a deadline for 3.11.3.Final and I didn't have time to change that, but I did it for 3.14.0.Final. So asking for the original behavior affects this class too.

Reply:
Okay. Maybe not a huge deal, but that's going to be different for EAP 7.3 and EAP 7.4 then. I'll have to back-port that fix to the 3.12 and 3.13 PR's I guess too.

Reply:
Yeah, my bad.